src/emoji/data.js

```diff
@@ -47,53 +49,40 @@ export const emojiTypeFromReactionType = (reactionType: ReactionType): EmojiType
  */
 export const getFilteredEmojis = (
   query: string,
-  activeImageEmojiByName: $ReadOnly<{| [string]: ImageEmojiType |}>,
-): $ReadOnlyArray<{| emoji_type: EmojiType, name: string, code: string |}> => {
-  // We start by making a map from matching emoji names to a number
-  // representing how good a match it is: 0 for a prefix match, 1 for a
-  // match anywhere else in the string.
-
-  const matchingUnicodeEmoji = objectEntries(unicodeCodeByName)
-    .map(([name, code]) => {
-      // This logic does not do any special handling for things like
-      // skin-tone modifiers or gender modifiers, since Zulip does not
-      // currently support those: https://github.com/zulip/zulip/issues/992.
-      // Once support is added for that, we may want to come back here and
-      // modify this logic, if for instance, there is a default skin-tone
-      // setting in the webapp that we want to also surface here. (or
-      // perhaps it will be best to leave it as is - that's a product
-      // decision that's yet to be made.) For the time being, it seems
-      // better to not show the user anything if they've searched for an
-      // emoji with a modifier than it is to show them the non-modified
-      // emoji, hence the very simple matching.
-      const matchesEmojiLiteral = parseUnicodeEmojiCode(code) === query;
-      const matchesEmojiName = Math.min(1, name.indexOf(query));
-      return [name, matchesEmojiLiteral ? 0 : matchesEmojiName];
-    })
-    .filter(([_, i]) => i !== -1);
+  activeImageEmoji: $ReadOnlyArray<EmojiForShared>,
+): $ReadOnlyArray<EmojiForShared> => {
+  const matcher = typeahead.get_emoji_matcher(query);
+  const allMatchingEmoji: Map<string, EmojiForShared> = new Map();
+  const unicodeEmojiObjects = objectEntries(unicodeCodeByName).map(([name, code]) => ({
+    emoji_type: 'unicode',
+    emoji_name: name,
+    emoji_code: code,
+  }));
+  for (const emoji of unicodeEmojiObjects) {
+    // This logic does not do any special handling for things like
```

Comment:
This comment is talking about the implementation details of how we match a literal emoji. So it should move to where those implementation details now live, and then get deleted from here.